includes/wikia/api/ArticlesApiController.class.php

```diff
@@ -1048,6 +1058,53 @@ public function getAsJson() {
 		$this->setResponseData( $result, '', self::SIMPLE_JSON_VARNISH_CACHE_EXPIRATION );
 	}
 
+	/**
+	 * Get all top level sections as an array by splitting on h2 tags
+	 * @param $content
+	 * @return array
+	 */
+	private function splitArticleIntoSections( $content ) {
+		$pattern = '/(\<h2[^\>]*>)/';
```

Comment:
I added the section attribute check. 

It's interesting to note that if you hard-code an `h2` tag in the article source, MW doesn't quite know what to do with it. See http://lizlux.liz.wikia-dev.com/wiki/Short_with_headers. The section is included in the TOC, but there's no edit link. However there is a `section` attribute added to the `h2` tag. Going to `?action=edit&section=X` does not count that section, and will either be empty or show the following section. 

For our purposes, I think this the current approach is fine. It will consider the offending `h2` to be a separate section but it's so edge-casey and the downside is so little that I wouldn't worry about it.